x/market/types/message_deal.go

```diff
@@ -71,3 +71,43 @@ func (msg *MsgCreateDeal) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{creator}
 }
+
+var _ sdk.Msg = &MsgSellData{}
+
+func NewSellData(cert DataValidationCertificate, seller string) *MsgSellData {
+	return &MsgSellData{
+		Cert:   &cert,
+		Seller: seller,
+	}
+}
+
+func (msg *MsgSellData) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgSellData) Type() string {
+	return "SellData"
+}
+
+// ValidateBasic is validation for MsgCreateDeal.
+func (msg *MsgSellData) ValidateBasic() error {
```

Comment:
Why don't we add the necessary field verification?

Reply:
Sorry for that. I will add ```ValidateBasic``` for verifying field. I'll put the ```TODO``` comment on this function.

Reply:
Please take out the comment that you're sorry. 😀
You don't have to be sorry :)